### PR TITLE
Rework implementation of `dpnp.degrees` and `dpnp.rad2deg` functions

### PR DESCRIPTION
**Pull request by antonwolfy**
The PR proposes to rework implementation of `dpnp.rad2deg` and `dpnp.degrees` functions which are quite limited now.
Both functions are completely equivalent to one another.

The implementation is done by adding new binary universal function `_degrees` to `_ufunc_impl` pybind11 extension.
The all affected tests are updated to cover different use cases.

- [x] Have you provided a meaningful PR description?
- [x] Have you added a test, reproducer or referred to issue with a reproducer?
- [x] Have you tested your changes locally for CPU and GPU devices?
- [x] Have you made sure that new changes do not introduce compiler warnings?
- [ ] Have you checked performance impact of proposed changes?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?

### CODE REVIEW

File: tests/test_umath.py [hunk 1]
```diff
@@ -279,6 +279,19 @@ def test_invalid_shape(self, shape):
             dpnp.copysign(dp_array, dp_array, out=dp_out)
 
 
+class TestDegrees:
+    @pytest.mark.parametrize(
+        "dtype", get_all_dtypes(no_none=True, no_complex=True)
+    )
+    def test_degrees(self, dtype):
+        a = numpy.array([numpy.pi, -0.5 * numpy.pi], dtype=dtype)
```
Comment (vlad-perevezentsev) [inline tests/test_umath.py:287]: Could you cover more cases for this function?
For example, `numpy.array([0])`, array with `nan` and `+-inf` and result with `large angles`

Comment (antonwolfy) [reply tests/test_umath.py:287]: `0` value is covered by `tests/test_usm_type.py::test_1in_1out` and `tests/test_sycl_queue.py::test_1in_1out` tests

Comment (antonwolfy) [reply tests/test_umath.py:288]: More tests are added

**Review by vlad-perevezentsev — APPROVED** [review]
LGTM!
Thank you @antonwolfy !

**Pull request merged**